python-client/giskard/models/model_explanation.py

```diff
@@ -66,13 +140,52 @@ def prediction_function(_df):
 
 
 def _get_highest_proba_shap(shap_values: list, model: BaseModel, dataset: Dataset) -> list:
-    """Get SHAP explanations of classes with the highest predicted probability."""
+    """Get SHAP explanations of classes with the highest predicted probability.
+
+    Parameters
+    ----------
+    shap_values : list (n_classes x n_samples)
+        Raw SHAP contributions of the features from the `dataset` to the `model`
+        predictions.
+
+    model : giskard.Model
+        Model, which output is explained using SHAP.
+
+    dataset : giskard.Dataset
+        Dataset, which entries are explained using SHAP.
+
+    Returns
+    -------
+    list
+        SHAP explanations of classes with the highest predicted probability.
+    """
     predictions = model.predict(dataset).raw_prediction
     return [shap_values[predicted_class][sample_idx] for sample_idx, predicted_class in enumerate(predictions)]
 
 
 def explain_with_shap(model: BaseModel, dataset: Dataset, only_highest_proba: bool = True) -> ShapResult:
-    """Get SHAP explanation result."""
+    """Get SHAP explanation result.
+
+    Return `ShapResult` object with SHAP explanations and related metadata.
```

Comment:
Explain the model with SHAP and return the results as a `ShapResult` object.

Reply:
\+